plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/params/GetResourceParams.java

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.client.params;
+
+import javax.validation.constraints.NotNull;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Arguments holder for {@link org.eclipse.che.plugin.docker.client.DockerConnector#getResource(GetResourceParams)}.
+ *
+ * @author Mykola Morhun
+ */
+public class GetResourceParams {
+
+    private String container;
+    private String sourcePath;
+
+    /**
+     * Creates arguments holder with required parameters.
+     *
+     * @param container
+     *         container id or name
+     * @param sourcePath
+     *         info about this parameter see {@link #withSourcePath(String)}
+     * @return arguments holder with required parameters
+     * @throws NullPointerException
+     *         if {@code container} or {@code sourcePath} is null
+     */
+    public static GetResourceParams from(@NotNull String container, @NotNull String sourcePath) {
+        return new GetResourceParams().withContainer(container)
+                                      .withSourcePath(sourcePath);
+    }
+
+    private GetResourceParams() {}
+
+    /**
+     * Adds container to this parameters.
+     *
+     * @param container
+     *         container id or name
+     * @return this params instance
+     * @throws NullPointerException
+     *         if {@code container} is null
+     */
+    public GetResourceParams withContainer(@NotNull String container) {
+        requireNonNull(container);
```

Comment:
NPE without any message is not developer friendly API. I think it is better to add message as second parameter in this and all other places like that.